test/credentials.spec.js

```diff
@@ -1899,6 +1899,51 @@
     });
   });
 
+  describe('AWS.AssumeRoleWithWebIdentityCredentials', function() {
+    var origEnv;
+    var fs = require('fs');
+    beforeEach(function() {
+      origEnv = process.env;
+      process.env = {
+        AWS_WEB_IDENTITY_TOKEN_FILE: 'envTokenFile',
+        AWS_IAM_ROLE_ARN: 'envRoleArn',
+        ENV_ROLE_SESSION_NAME: 'envSessionName'
+      };
+      helpers.spyOn(AWS.util, 'readFileSync').andReturn(
+        '[default]\nweb_identity_token_file = cfgTokenFile\nrole_arn = cfgRoleArn\nrole_session_name = cfgSessionName'
+      );
```

Comment:
The SDK caches the config file content. So before mocking the fs read file function, make sure to evict the cache like [here](https://github.com/aws/aws-sdk-js/blob/2627d1af01bf1a1dc745361c3edf9e825594dcd4/test/credentials.spec.js#L253)

Reply:
Resolved in https://github.com/aws/aws-sdk-js/pull/2737/commits/bc5084df4b38259165612f969b842484d5f80b91